File: Backup/app/schemas/request/text.py
from pydantic import BaseModel, Field, validator, constr
from typing import Literal
from enum import Enum


class TextAction(str, Enum):
    json_format = "json_format"
    base64_encode = "base64_encode"
    base64_decode = "base64_decode"
    url_encode = "url_encode"
    url_decode = "url_decode"


class TextProcessRequest(BaseModel):
    text: constr(min_length=1, max_length=100000) = Field(...,
                                                          description="要处理的文本内容")
    action: TextAction = Field(..., description="操作类型")

    @validator('text')
    def validate_text_not_empty(cls, v):
        if not v or not v.strip():
            raise ValueError("文本内容不能为空")
        return v


class TextCompareRequest(BaseModel):
    text1: constr(min_length=0, max_length=100000) = Field(...,
                                                           description="第一个文本")
    text2: constr(min_length=0, max_length=100000) = Field(...,
                                                           description="第二个文本")

File: Backup/tools_project_rust/src/models/text.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum TextAction {
    JsonFormat,
    Base64Encode,
    Base64Decode,
    UrlEncode,
    UrlDecode,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct TextProcessRequest {
    pub text: String,
    pub action: TextAction,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct TextProcessResponse {
    pub result: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct TextCompareRequest {
    pub text1: String,
    pub text2: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct LineDifference {
    pub line: usize,
    pub text1: String,
    pub text2: String,
    #[serde(rename = "type")]
    pub diff_type: String,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct CompareSummary {
    pub total_lines: usize,
    pub different_lines: usize,
    pub identical: bool,
}

#[derive(Debug, Serialize, Deserialize)]
pub struct TextCompareResponse {
    pub differences: Vec<LineDifference>,
    pub summary: CompareSummary,
}
